examples/opengl/OpenGL.cpp

```diff
@@ -75,10 +78,14 @@ int main()
         // Attempt to generate a mipmap for our cube texture
         // We don't check the return value here since
         // mipmapping is purely optional in this example
-        texture.generateMipmap();
+        (void) texture.generateMipmap();
```

Comment:
```cpp
static_cast<void>(texture.generateMipmap());
```

Reply:
Do we have no warning enabled for C-casts? 🤔

Reply:
Eh, `(void)` is ubiquitous. It's not really a bad use of a C-style cast IMHO.

Reply:
With @eXpl0it3r's remark in mind, it's probably still good to use `static_cast`. Do we want to enable CI lints for C-style and function-style casts? Although we would need to limit it to SFML code itself (not extlibs)...

I agree that C++ chose the worst possible cast syntax of all languages, but well...

The alternative would be something like `SFML_UNUSED(...)`, but not sure if that's much nicer. I'd say casting to void is commonly understood.

Reply:
I disagree, as I mentioned everyone knows that `(void) foo;` means "discard `foo`", and there is absolutely no risk in using that syntax in modern C++. I hate C as much as everyone else, but using `static_cast` here is needless. I feel a bit in the same way you feel about me trying to shoehorn `[[nodiscard]]` into everything :)

By the way, C-style cast warnings **intentionally exclude** `(void)` casts. See https://stackoverflow.com/a/53485874/598696

Reply:
> I feel a bit in the same way you feel about me trying to shoehorn [[nodiscard]] into everything :)

So it's revenge! 😁

Jokes aside, if `(void) expr` is really a common idiom and is explicitly whitelisted by compilers, that changes things of course. I just checked with VS 2019, seems to be the same. Warning C4189 is not emitted with `(void) expr`, on level /W4.

Reply:
Just FYI: we already have `-Wold-style-cast` activated, see [here](https://github.com/SFML/SFML/blob/master/cmake/CompilerWarnings.cmake#L46). Like already mentioned, because it is so ubiquitous, `(void)` doesn't trigger the warning.